motoman_hc10_support/urdf/hc10dt_b10_macro.xacro

```diff
@@ -141,12 +140,23 @@
       <axis xyz="-1 0 0" />
       <limit lower="${radians(-180)}" upper="${radians(180)}" effort="31.36" velocity="${radians(250)}"/>
     </joint>
-    <joint name="${prefix}joint_6_t-tool0" type="fixed">
+    <!-- end of joint list -->
+
+    <!-- ROS-Industrial 'flange' frame: attachment point for EEF models -->
+    <link name="${prefix}flange"/>
+    <joint name="${prefix}joint_6_t-flange" type="fixed">
       <origin xyz="0.170 0 0" rpy="${radians(180)} ${radians(-90)} 0"/>
```

Comment:
Is this correct?

`flange` should have X+ forward. Is that the case?

(I haven't checked in RViz)

Reply:
The HC10DT-B10 changed the home position as shown in picture further up.  The flange is pointing forward, so the Z+ is pointing forward.  This should allow to add tools (end-effectors) with angles that matches the values in controller tool defintions.

Reply:
It is the same as the GP4
![Screenshot from 2022-01-06 14-31-22](https://user-images.githubusercontent.com/4691677/148440567-fe439d2b-f20d-4f88-b153-4580a620b652.png)
4

Reply:
There is a difference between `tool0` and `flange`, and it has to do with their intended use.

`tool0` is supposed to coincide with an "all-zeros toolframe", as the Yaskawa controller defines it. It just happens that such a tool has Z+ "forward" (ie: away from the face plate).

`flange` is supposed to be used to attach EEF models in a composite `.xacro`. To make this as straightforward as possible, `flange` should be oriented such that it adheres to the "normal ROS frame orientation" (as defined in REP-103), which has X+ pointing forward. This is a link-local frame, and as such, "forward" (and the other cardinalities) are defined relative to that link.

`tool0` is not intended to attach any EEF model to, nor a toolframe (ie: an additional TF frame used as a toolframe on the ROS side).

If I interpret the left screenshot correctly, `flange` and `tool0` have the same orientation there, being Z+ forward. The GP4 model also has that.

Both need to be fixed to follow conventions.

I'll submit a PR for the GP4 later.

Reply:
Yes, it fine with me.

Reply:
Resolved in 8e673ac.